src/diffusers/models/dualencoder_gfn.py

```diff
@@ -0,0 +1,695 @@
+# Model adapted from GeoDiff https://github.com/MinkaiXu/GeoDiff
+from typing import Callable, Dict, Union
+
+import numpy as np
+import torch
+import torch.nn.functional as F
+from torch import Tensor, nn
+from torch.nn import Embedding, Linear, Module, ModuleList, Sequential
+
+from rdkit.Chem.rdchem import BondType as BT
+from torch_geometric.nn import MessagePassing, radius, radius_graph
+from torch_geometric.typing import Adj, OptPairTensor, OptTensor, Size
+from torch_geometric.utils import dense_to_sparse, to_dense_adj
+from torch_scatter import scatter_add, scatter_mean
+from torch_sparse import SparseTensor, coalesce
+
+from ..configuration_utils import ConfigMixin
+from ..modeling_utils import ModelMixin
+
+
+BOND_TYPES = {t: i for i, t in enumerate(BT.names.values())}
```

Comment:
It seems like we only need the lenght of this dict - could we maybe instead hardcode this value here or have it in the model config instead of having to rely on a dependency for this? I'm not sure it's worth it to add an additional (slighly dangerous looking dependency) just to get the length of `BOND_TYPES` - what do you think?

Reply:
Oh interesting. I think this is all only used in training or in a different mode than the colab I built, I can delete this and a lot of other code (according to my tests, I will verify in the colab later).

Reply:
SO many deletions!